cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/PodRevision.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.resource;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.strimzi.api.kafka.model.StrimziPodSet;
+import io.strimzi.operator.common.Annotations;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.ReconciliationLogger;
+import io.strimzi.operator.common.Util;
+import io.strimzi.operator.common.model.Labels;
+
+/**
+ * In the future, as we better utilize the StrimziPodSet possibilities and not just replace StatefulSets 1-to-1, the
+ * revision might require more complicated setup. That is why this is using separate class, although it currently seems
+ * a bit simple.
+ */
+public class PodRevision {
+    public static final String STRIMZI_REVISION_ANNOTATION = Labels.STRIMZI_DOMAIN + "revision";
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(PodRevision.class.getName());
+    private final static ObjectMapper MAPPER = new ObjectMapper();
+
+    /**
+     * Generates the revision of the Pod. Currently, it just serializes the Pod using Jackson ObjectMapper and creates
+     * a SHA1 hashstub from it.
+     *
+     * @param reconciliation    Reconciliation identifier (used for logging)
+     * @param pod               Pod for which the revision should be calculated
+     *
+     * @return                  The revision string
+     */
+    public static String getRevision(Reconciliation reconciliation, Pod pod) {
+        try {
+            return Util.sha1Prefix(MAPPER.writeValueAsString(pod));
+        } catch (JsonProcessingException e) {
+            LOGGER.warnCr(reconciliation, "Failed to get pod revision", e);
+            throw new RuntimeException("Failed to get pod revision", e);
+        }
+    }
+
+    /**
+     * Compares the current Pod with the desired StrimziPodSet to decide if the desired pod changed and needs to be rolled.
+     * It uses the revision to detect the changes. The desired revision is compared with the current revision.
+     *
+     * @param currentPod        Current pod
+     * @param desiredPodSet    The StrimziPodSet resource with the desired Pods
+     *
+     * @return                  True if the revision changed. False otherwise.
+     */
+    public static boolean hasChanged(Pod currentPod, StrimziPodSet desiredPodSet)   {
+        Pod desiredPod = desiredPodSet
+                .getSpec()
+                .getPods()
+                .stream()
+                .map(pod -> MAPPER.convertValue(pod, Pod.class))
```

Comment:
It is not used because it cannot be translated into CRD with our `crd-generator` and writing our custom model for the whole Pod structure seemed unnecessary. The only thing we loose by this is validation -> that is anyway semi-functional and not so important given it is an internal resource only.